### PR TITLE
Pcre

### PR DESCRIPTION
**Pull request by barcharcraz**
add the pcre library.

### CODE REVIEW

File: ports/pcre/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    SHA512 82f1c2bdd0a6cc086e3734621ac7a2773cb28f42cf5e400b9bbe8c16655465d9367bce82c6db69577c40ec137b30f1b2443a8d91998a514f81e1c2210828a113
+)
+vcpkg_extract_source_archive(${ARCHIVE})
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    OPTIONS -DPCRE_BUILD_TESTS=NO
+            -DPCRE_BUILD_PCREGREP=NO
+            -DPCRE_BUILD_PCRE32=YES
+            -DPCRE_BUILD_PCRE16=YES
```
Comment (Orvid) [inline ports/pcre/portfile.cmake]: Could you add `PCRE_SUPPORT_JIT=YES`, `PCRE_SUPPORT_UNICODE_PROPERTIES=YES` and `PCRE_SUPPORT_UTF=YES`? They all default to off, but I'll need all of them for my use-case.

Also, you'll want to set `INSTALL_MSVC_PDB=YES` as well.

Comment (barcharcraz) [inline ports/pcre/portfile.cmake]: INSTALL_MSVC_PDB caused problems, may as well just copy the pdbs with vcpkg_copy_pdbs()

I'll add those other options though, you're right, they should be there!

Comment (alexkaratarakis) [inline ports/pcre/portfile.cmake]: @barcharcraz Are the flags added in the second commit encompassing the ones mentioned by @Orvid ?

Comment (barcharcraz) [inline ports/pcre/portfile.cmake]: no, I just pushed them

**Comment by ras0219-msft** [comment]
Thanks!

**Pull request merged**